api_app/utility.py

```diff
@@ -0,0 +1,191 @@
+        # gen new task_id
+        task_id = uuid()
+        # construct arguments
+        args = [
+            job_id,
+            config.asdict(),
+            {"runtime_configuration": runtime_params, "task_id": task_id},
+        ]
+        # get celery queue
+        queue = config.config.queue
+        if queue not in settings.CELERY_QUEUES:
+            logger.warning(
+                f"Analyzer {a_name} has a wrong queue." f" Setting to `{DEFAULT_QUEUE}`"
+            )
+            queue = DEFAULT_QUEUE
+        # get soft_time_limit
+        soft_time_limit = config.config.soft_time_limit
+        # create task signature and add to list
+        task_signatures.append(
+            tasks.run_analyzer.signature(
+                args,
+                {},
+                queue=queue,
+                soft_time_limit=soft_time_limit,
+                task_id=task_id,
+            )
+        )
+        analyzers_used.append(a_name)
+
+    return task_signatures, analyzers_used
+
+
+def stack_connectors(
+    job_id: int,
+    connectors_to_execute: List[str],
+    runtime_configuration: Dict[str, Dict] = None,
+) -> Tuple[List[Signature], List[str]]:
+    from intel_owl import tasks
+
+    # to store the celery task signatures
+    task_signatures = []
+
+    connectors_used = []
+    connectors_to_run = runnable_connectors(connectors_to_execute)
+
+    # get connectors config
+    connector_dataclasses = ConnectorConfig.filter(names=connectors_to_run)
+
+    # loop over and create task signatures
+    for c_name, cc in connector_dataclasses.items():
+        # if disabled or unconfigured (this check is bypassed in STAGE_CI)
+        if not settings.STAGE_CI:
+            continue
+
+        # get runtime_configuration if any specified for this analyzer
+        runtime_params = runtime_configuration.get(c_name, {})
+        # gen a new task_id
+        task_id = uuid()
+        # construct args
+        args = [
+            job_id,
+            cc.asdict(),
+            {"runtime_configuration": runtime_params, "task_id": task_id},
+        ]
+        # get celery queue
+        queue = cc.config.queue
+        if queue not in settings.CELERY_QUEUES:
+            logger.error(
+                f"Connector {c_name} has a wrong queue."
+                f" Setting to `{DEFAULT_QUEUE}`"
+            )
+            queue = DEFAULT_QUEUE
+        # get soft_time_limit
+        soft_time_limit = cc.config.soft_time_limit
+        # create task signature and add to list
+        task_signatures.append(
+            tasks.run_connector.signature(
+                args,
+                {},
+                queue=queue,
+                soft_time_limit=soft_time_limit,
+                task_id=task_id,
+                ignore_result=True,  # since we are using group and not chord
+            )
+        )
+        connectors_used.append(c_name)
+    
+    return task_signatures, connectors_used
```

Comment:
I'm thinking how we can remove this lines from there, like i said before I really hate having a "dump" file where we put functions that we do not where to put. This is prone to having a "god utilities" where the file size will increase exponentially. Once I have a solution, i will comment this

Reply:
Okay sure. I will ask for more opinions from matteo then.

Reply:
if the file would have grown, it would have been a hard file to get back to with a lot of junk. i appreciate you pointing out how it could be further cleaned up.

Reply:
Unfortunately I can see the decision on why these methods should be on a common file between the two callers. I'm taking a bit of time to decide on how to structure this, I do not like to give suggestion and then understand that the suggestion itself was shit